actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -87,6 +87,7 @@ module ClassMethods
       #
       # * <tt>:only/:except</tt> - Passed to the <tt>before_action</tt> call. Set which actions are verified.
       # * <tt>:if/:unless</tt> - Passed to the <tt>before_action</tt> call. Set when actions are verified.
+      # * <tt>:prepend</tt> - ?
```

Comment:
:prepend - By default, the verification of the authentication token is added to the front of the callback chain. If you need to make the verification depend on other callbacks, like authentication methods (say cookies vs oauth), this might not work for you. Pass prepend: false to just add the verification callback in the position of the protect_from_forgery call. This means any callbacks added before are run first.